pkg/kubelet/cm/topologymanager/numa_info.go

```diff
@@ -0,0 +1,127 @@
+package topologymanager
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
+)
+
+const (
+	defaultNodeDir = "/sys/devices/system/node/"
+)
+
+var sysFs = &SysFs{nodeDir: defaultNodeDir}
+
+type Distances [][]uint64
+
+type NUMAInfo struct {
+	Nodes     []int
+	Distances Distances
+}
+
+func NewNUMAInfo(topology []cadvisorapi.Node) (*NUMAInfo, error) {
+	var numaNodes []int
+	distances := make([][]uint64, len(topology))
+	for _, node := range topology {
+		numaNodes = append(numaNodes, node.Id)
+
+		// uncomment when this commit is released https://github.com/google/cadvisor/commit/24dd1de08a72cfee661f6178454db995900c0fee
+		// distances[node.Id] = append(distances[node.Id], node.Distances...)
+
+		// for now we need to retrieve this information in Kubelet
+		nodeDistance, err := sysFs.GetDistances(node.Id)
+		if err != nil {
+			return &NUMAInfo{}, err
+		}
+		distances[node.Id] = nodeDistance
+
+	}
+
+	return &NUMAInfo{
+		Nodes:     numaNodes,
+		Distances: distances,
+	}, nil
+}
+
+func (n *NUMAInfo) Narrowest(m1 bitmask.BitMask, m2 bitmask.BitMask) bitmask.BitMask {
+	if m1.IsNarrowerThan(m2) {
+		return m1
+	}
+
+	return m2
+}
+
+func (n *NUMAInfo) Closest(m1 bitmask.BitMask, m2 bitmask.BitMask) bitmask.BitMask {
+	if m1.Count() == m2.Count() {
+		m1Distance := n.Distances.CalculateAverageFor(m1)
+		m2Distance := n.Distances.CalculateAverageFor(m2)
+		// if average distance is the same take bitmask with more lower-number bits set
+		if m1Distance == m2Distance {
+			if m1.IsLessThan(m2) {
+				return m1
+			}
+			return m2
+		} else if m1Distance < m2Distance {
+			// m1 avg distance is lower
+			return m1
+		}
+
+		// m2 avg distance is lower
+		return m2
```

Comment:
Instead of 2 separate comments, can you put one comment above this second if condition, something like:
```
// Otherwise, return the bitmask with the shortest average distance between NUMA nodes
...
```